src/components/calculator.js

```diff
@@ -1,71 +1,63 @@
 import '../stylesheets/calculator.css';
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import calculate from '../logic/calculate';
 
-class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
+const Calculator = () => {
+  const [state, calcState] = useState({
+    total: 0,
+    next: 0,
+    operation: '',
+  });
 
-    this.state = {
-      total: 0,
-      next: 0,
-      operation: '',
-    };
-    this.handleClick = this.handleClick.bind(this);
-  }
-
-  componentDidUpdate() {
-    const { total, next } = this.state;
+  useEffect(() => {
+    const { total, next } = state;
     const inputEl = document.querySelector('input');
 
     if ((next != null)) {
       inputEl.value = next;
     } else if (!Number.isNaN(total)) {
       inputEl.value = total;
     }
-  }
+  });
```

Comment:
**Optional Change**
- [ ] The useEffect hook takes in two parameters, a function and an array of dependencies that will allow the component only re-render when they are updated. Currently you haven't passed any dependencies nor included the empty array, meaning your component will unnecessarily re-render anyway. Let us pass the empty array or dependencies in the array to maximize React's performance.